src/repr/src/row.rs

```diff
@@ -2545,11 +2524,12 @@ impl RowPacker<'_> {
 
     /// Truncates the underlying row to contain at most the first `n` datums.
     pub fn truncate_datums(&mut self, n: usize) {
+        let prev_len = self.row.data.len();
         let mut iter = self.row.iter();
         for _ in iter.by_ref().take(n) {}
-        let offset = iter.offset;
+        let next_len = self.row.data.len();
```

Comment:
I think this is what we need:
```suggestion
        let next_len = iter.data.len();
```